src/dotnet/Rider.Plugins.EfCore/Extensions/PsiExtensions.cs

```diff
@@ -63,5 +70,15 @@ public static IEnumerable<IClass> FindInheritorsOf(this IPsiModule module, IProj
             @class
                 .GetAttributeInstances(AttributesSource.All)
                 .SingleOrDefault(attribute => attribute.GetAttributeShortName() == attributeShortName);
+
+        private static string GetClassFolderAbsPath(this IClass @class, string className)
+        {
+            var result = @class.GetSourceFiles()
+                .FirstOrDefault(x => x.Name == className);
```

Comment:
Is it necessary to find some specific file in the collection of the source files of a class?

Reply:
This solution has came to my mind after not so deep research in `JetBrains.ReSharper.Psi` namespace. It cannot be considered as grace or optimal one. Please advice on better solution

Reply:
`IClass.GetSourceFiles()` returns source files related to class implementation. In the case of migration classes, it will be the main implementation file and a designer file. Both of them are always generated in the same folder, so just using `FirstOrDefault()` without condition will be enough.